src/java.sql/share/classes/java/sql/Connection.java

```diff
@@ -1676,5 +1676,268 @@ default void setShardingKey(ShardingKey shardingKey, ShardingKey superShardingKe
     default void setShardingKey(ShardingKey shardingKey) throws SQLException {
         throw new SQLFeatureNotSupportedException("setShardingKey not implemented");
     }
+    // JDBC 4.5
 
+    /**
+     * Returns a {@code String} enclosed in single quotes. Any occurrence of a
+     * single quote within the string will be replaced by two single quotes.
+     *
+     * <blockquote>
+     * <table class="striped">
+     * <caption>Examples of the conversion:</caption>
+     * <thead>
+     * <tr><th scope="col">Value</th><th scope="col">Result</th></tr>
+     * </thead>
+     * <tbody style="text-align:center">
+     * <tr> <th scope="row">Hello</th> <td>'Hello'</td> </tr>
+     * <tr> <th scope="row">G'Day</th> <td>'G''Day'</td> </tr>
+     * <tr> <th scope="row">'G''Day'</th>
+     * <td>'''G''''Day'''</td> </tr>
+     * <tr> <th scope="row">I'''M</th> <td>'I''''''M'</td>
+     * </tr>
+     *
+     * </tbody>
+     * </table>
+     * </blockquote>
+     * @implSpec
+     * The default implementation creates the literal as:
+     * {@code "'" + val.replace("'", "''") + "'"}.
+     * @implNote
+     * JDBC driver implementations may need to provide their own implementation
+     * of this method in order to meet the requirements of the underlying
+     * datasource.
```

Comment:
I realize this is how Statement.enquoteLiteral is specified since its introduction in JDK 9 but I can't see how a JDBC driver has any leeway to effect a different outcome. The spec is very clear that it quotes the input and replaces any single quotes with two single quotes. Do you have any examples where a "requirement" from a datasource could result in a different outcome? I could imagine a driver implementation doing its own implementation for performance reasons but not sure beyond that.